### ISSUE TITLE
[Proposal] HTTP/2 Client APIs

### ISSUE THREAD

**Issue by joshaustintech** [java/devjava-content#74]
### Request Issue

#8

### Website Section

Home --> Tutorials --> HTTP Client

### Proposal Details

- Why this API and not other HTTP clients?
    - Lightweight
    - No dependencies 
- Fetching recipe data from [TheMealDB](https://www.themealdb.com)
  - Building an `HttpClient`
  - Building an `HttpRequest`
  - Getting an `HttpResponse`
    - Method 1: synchronous
    - Method 2: asynchronous

### Author References

- https://joshaustin.tech/blog/how-to-concatenate-strings-in-java-like-a-pro/
- https://joshaustin.tech/blog/getting-started-with-rife2/
- https://joshaustin.tech/blog/those-sneaky-nullpointerexceptions/

**Comment by joshaustintech** [comment]
Hi @carimura, @nipafx, I'm new to this process. Anything I you need from me?

**Comment by carimura** [comment]
Thanks for the submission and sorry for delay, I've been on paternity leave. Looks straightforward. "Why this API" seems good but I would avoid going too far into the "why not other HTTP clients".  The rest seems fun. 

Thoughts @nipafx @JosePaumard ?

**Comment by joshaustintech** [comment]
@carimura congrats on the new arrival!

**Comment by nipafx** [comment]
Hey @joshaustintech , thanks for the proposal. It looks good! Some comments:

* The website section should be https://dev.java/learn/ -> _Mastering the API_
* I recommend to start with the tutorial (your second bullet point) and come to reasons for using this API (your first bullet point) later - this way the article starts with what people are most likely to be searching for.
* When discussing the synchronous approach, please bring up virtual threads and link to [the respective article](https://dev.java/learn/new-features/virtual-threads/).
* Let's not use MealDB. I'd have to take a closer look at the website to verify that it can feature in our tutorials. Instead, let's use dev.java or inside.java.
* In case you need inspiration, feel free to check the two posts I wrote on this API: [1](https://nipafx.dev/java-http-2-api-tutorial/), [2](https://nipafx.dev/java-reactive-http-2-requests-responses/)

Feel free to open a PR once you've got something to review. Note that I'll be out of office unitl early May, though, so no need to rush. 😉

**Comment by joshaustintech** [comment]
Sounds great. Thanks @nipafx. 😎

**Comment by carimura** [comment]
Hi @joshaustintech just going through open issues -- did you want to open a PR for this? We are happy to review as soon as that's in. and thanks!

**Comment by joshaustintech** [comment]
Yes, I would love to open a PR soon as I have the content ready in the next few weeks.

**Comment by joshaustintech** [comment]
Thank you all for your patience, still planning to open a PR for this soon.
